src/actions/thunks/putItemFactory.test.ts

```diff
@@ -0,0 +1,89 @@
+import { putItemFactory } from './putItemFactory';
+import { ITEM_SAVE_FAIL, ITEM_SAVE_REQUEST, ITEM_SAVE_SUCCESS } from '../types/listActionTypes';
+import { IListAction } from '../types/IListAction';
+import { Dispatch } from '../types/Dispatcher';
+import Mock = jest.Mock;
+
+describe('Put item', () => {
+  const dispatch: Mock<Dispatch<IListAction>> = jest.fn();
+  const itemId = '00000000-0000-0000-0000-000000000001';
+  const itemText = 'updated';
+  const fetchOk = jest.fn(() => Promise.resolve({
+    status: 200,
+    statusText: 'OK',
+    json: () => ({id: itemId, text: itemText}),
+    ok: true,
+  }));
+
+  const fetchNotFound = jest.fn(() => Promise.resolve({
+    status: 404,
+    statusText: 'Not Found',
+  }));
+
+  const fetchBadRequest = jest.fn(() => Promise.resolve({
+    status: 400,
+    statusText: 'Bad Request',
+  }));
+
+  const invalidFetchCases = [fetchNotFound, fetchBadRequest];
+  const allFetchCases = [...invalidFetchCases, fetchOk];
+
+  beforeEach(() => {
+    dispatch.mockClear();
+  });
+
+  it('dispatches request and success actions if the fetch response was successful', () => {
+    return putItemFactory(fetchOk)(itemId, itemText)(dispatch).then(() => {
```

Comment:
Split this per closure, something like this would be more readable
```suggestion
    const putItem = putItemFactory(fetchOk);
    const dispatchable = putItem(itemId, itemText);
    return dispatchable(dispatch);
```
You may even consider rewritting tests with async/await 😈 
Applies for all thunks